migrations/capcons/migration_test.go

```diff
@@ -1586,3 +1592,111 @@ func TestLinkMigration(t *testing.T) {
 		test(linkTestCase)
 	}
 }
+
+func TestClearPrivateDomain(t *testing.T) {
+
+	t.Parallel()
+
+	pathLinks := []testLink{
+		{
+			// Equivalent to:
+			//   link<&Test.R>(/public/test1, target: /private/test2)
+			sourcePath: interpreter.PathValue{
+				Domain:     common.PathDomainPublic,
+				Identifier: "test1",
+			},
+			targetPath: interpreter.PathValue{
+				Domain:     common.PathDomainPrivate,
+				Identifier: "test2",
+			},
+			borrowType: testRReferenceStaticType,
+		},
+		{
+			// Equivalent to:
+			//   link<&Test.R>(/private/test3, target: /storage/test4)
+			sourcePath: interpreter.PathValue{
+				Domain:     common.PathDomainPrivate,
+				Identifier: "test3",
+			},
+			targetPath: interpreter.PathValue{
+				Domain:     common.PathDomainStorage,
+				Identifier: "test4",
+			},
+			borrowType: testRReferenceStaticType,
+		},
+	}
+
+	accountLinks := []interpreter.PathValue{
+		// Equivalent to:
+		//   linkAccount(/public/test5)
+		{
+			Domain:     common.PathDomainPublic,
+			Identifier: "test5",
+		},
+		// Equivalent to:
+		//   linkAccount(/private/test6)
+		{
+			Domain:     common.PathDomainPrivate,
+			Identifier: "test6",
+		},
+	}
+
+	rt := NewTestInterpreterRuntime()
+
+	runtimeInterface := &TestRuntimeInterface{
+		Storage: NewTestLedger(nil, nil),
+	}
+
+	// Create and store path and account links
+
+	storage, inter, err := rt.Storage(runtime.Context{
+		Interface: runtimeInterface,
+	})
+	require.NoError(t, err)
+
+	storeTestPathLinks(t, pathLinks, storage, inter)
+
+	storeTestAccountLinks(accountLinks, storage, inter)
+
+	err = storage.Commit(inter, false)
+	require.NoError(t, err)
```

Comment:
Maybe also assert the size of the private domain storage before the migration

Reply:
Good idea! Added in 25fdcb41d3c5779b679431d8cb0fc14446ee419e